__tests__/components/Table.spec.js

```diff
@@ -361,6 +361,81 @@ describe('Table component', () => {
 
       expect(cellsText).toEqual(dataText)
     })
+
+    it('should update the table when data changes', () => {
+      const field = 'Name'
+      const cols = [
+        {
+          field
+        }
+      ]
+
+      const data = [
+        {
+          Name: 'Caterpie',
+          description: 'lorem ipsum dolor sit amet'
+        },
+        {
+          Name: 'Blastoise',
+          description: 'consecteur adipiscing elit, sed do eiusmod'
+        }
+      ]
+
+      const component = mount(
+        <Table columns={cols} data={data} />
+      )
+
+      data.push({
+        Name: 'Squrtle'
+      })
+
+      component.setProps({ data })
+
+      const cellsText = component.find('[data-display="table-body"] [data-display="table-cell"]')
+        .map(element => element.text())
+
+      const dataText = _.map(data, field)
+
+      expect(cellsText).toEqual(dataText)
```

Comment:
How about also modifying an existing item? We could check:
* "modifying" the array in an immutable way (like we do in react) to set a new `Name`.
* replacing the array with `_.cloneDeep()` could let us check the respective `_.isEqual()` check in the row component. I'm not sure whether we can actually test that the table did a re-render though. Oh maybe by spying on the `render()` function that we provide. This would surface regressions to that mechanism :)
* We might also want to add a failing test where we modify one of the the array elements in a mutable way. This could surface regressions that would cause unnecessary re-renders of the table.